src/spartan_app.erl

```diff
@@ -124,11 +126,19 @@ load_json_config(FileBin) ->
 
 process_config_tuple({<<"upstream_resolvers">>, UpstreamResolvers}) ->
     UpstreamResolverIPs = lists:map(fun parse_ipv4_address/1, UpstreamResolvers),
-    ConfigValue = [{UpstreamResolverIP, 53} || UpstreamResolverIP <- UpstreamResolverIPs],
+    UpstreamResolverPorts = lists:map(fun parse_port_upstream_resolvers/1, UpstreamResolvers),
+    ConfigValue = [{UpstreamResolverIP, UpstreamResolverPort} || UpstreamResolverIP <- UpstreamResolverIPs, UpstreamResolverPort <-
+        UpstreamResolverPorts],
```

Comment:
I hadn't gotten a chance to review the code yet. If you'd like to take this PR over (create a new PR, same feature), I'd more than happy with that.